include/cinatra/coro_http_client.hpp

```diff
@@ -222,15 +222,48 @@ class coro_http_client {
       }
       else {
         if (has_closed_) {
-          if (ec = co_await asio_util::async_connect(
-                  io_ctx_, socket_, u.get_host(), u.get_port());
-              ec) {
-            break;
+          if (!proxy_host_.empty() && proxy_port_ != -1) {
+            if (ec = co_await asio_util::async_connect(
+                    io_ctx_, socket_, proxy_host_, std::to_string(proxy_port_));
+                ec) {
+              break;
+            }
           }
-          has_closed_ = false;
+          else {
+            if (ec = co_await asio_util::async_connect(
+                    io_ctx_, socket_, u.get_host(), u.get_port());
+                ec) {
+              break;
+            }
+            // Should the connection be actively closed in proxy? Maybe it's
+            // better to let the user judge.
+            has_closed_ = false;
+          }
+        }
+
+        std::string write_msg = "";
+        if (!proxy_host_.empty() && proxy_port_ != -1) {
+          std::string proxy_url = "";
+          if (u.get_port() == "http") {
+            proxy_url += " http://" + u.get_host() + ":";
+            proxy_url += "80";
+          }
+          else if (u.get_port() == "https") {
+            proxy_url += " https://" + u.get_host() + ":";
+            proxy_url += "443";
+          }
+          else {
+            // all be http
+            proxy_url += " http://" + u.get_host() + ":";
```

Comment:
these logic can be put into handle_uri function, and the main logic keep the same with before